client/state/comments/selectors.js

```diff
@@ -0,0 +1,118 @@
+
+/***
+ * Get total number of comments on the server for a given post
+ * @param {Object} state redux state
+ * @param {Object} props selector props
+ * @param {Number} props.siteId site identification
+ * @param {Number} props.postId site identification
+ * @return {Number} total comments count on the server
+ */
+export const getPostTotalCommentsCount = ( state, props ) => state.comments.totalCommentsCount.get( getCommentParentKey( props.siteId, props.postId ) );
+
+/***
+ * Get most recent comment date for a given post
+ * @param {Object} state redux state
+ * @param {Object} props selector props
+ * @param {Number} props.siteId site identification
+ * @param {Number} props.postId site identification
+ * @return {Date} most recent comment date
+ */
+export const getPostMostRecentCommentDate = createSelector(
+	getPostCommentItems,
+	( items ) => items && items.first() ? new Date( items.first().get( 'date' ) ) : undefined
+);
+
+/***
+ * Get most old (earliest) comment date for a given post
+ * @param {Object} state redux state
+ * @param {Object} props selector props
+ * @param {Number} props.siteId site identification
+ * @param {Number} props.postId site identification
+ * @return {Date} earliest comment date
+ */
+export const getPostOldestCommentDate = createSelector(
+	getPostCommentItems,
+	( items ) => items && items.last() ? new Date( items.last().get( 'date' ) ) : undefined
+);
+
+/***
+ * Gets comment tree for a given post
+ * @param {Object} state redux state
+ * @param {Object} props selector props
+ * @param {Number} props.siteId site identification
+ * @param {Number} props.postId site identification
+ * @return {Immutable.Map<CommentId, CommentNode>} comments tree in the form of immutable map<CommentId, CommentNode>, and in addition a children array
+ */
+export const getPostCommentsTree = createSelector(
+	getPostCommentItems,
+	( items ) => items ? buildCommentsTree( items ) : undefined
+);
+
+/***
+ * Whether we have more comments to fetch for a given post
+ * @param {Object} state redux state
+ * @param {Object} props selector props
+ * @param {Number} props.siteId site identification
+ * @param {Number} props.postId site identification
+ * @return {Boolean} do we have more comments to fetch
+ */
+export const haveMoreCommentsToFetch = createSelector(
+	getPostCommentItems,
+	getPostTotalCommentsCount,
+	( items, totalCommentsCount ) => items && totalCommentsCount ? items.size < totalCommentsCount : undefined
+);
+
+/***
+ * Gets likes stats for the comment
+ * @param {Object} state redux state
+ * @param {Object} props selector props
+ * @param {Number} props.siteId site identification
+ * @param {Number} props.postId site identification
+ * @param {Number} props.commentId comment identification
+ * @return {Immutable.Map} that has i_like and like_count props
+ */
+export const getCommentLike = createSelector(
+	getPostCommentItems,
+	( state, props ) => props.commentId,
```

Comment:
@gwwar Not really sure what you mean.
That's taken from [their example](https://github.com/reactjs/reselect#accessing-react-props-in-selectors). 
```javascript
const maxTodosSelector = (state, props) => props.maxTodos
const visibleTodosSelector = createSelector(
  ...
  maxTodosSelector,
  (visibilityFilter, todos, maxTodos) => {
   ...
  }
)
```